### PR TITLE
Upgrade to Containerd v1.5.7

### PR DESCRIPTION
**Pull request by ajayk**
- Upgrade to containerd v1.5.7 
- Resolves issue #214 

## Description

Updated references to Containerd Based on the previous pr noted in the #214 

## Motivation and Context

- [ ] I have raised an issue to propose this change **this is required**

## How Has This Been Tested?

Local build on ubuntu by running make and running the faasd locally 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

Commits:

- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] My commit message has a body and describe how this was tested and why it is required.
- [x] I have signed-off my commits with `git commit -s` for the Developer Certificate of Origin (DCO)

Code:

- [ ] My code follows the code style of this project.
- [ ] I have added tests to cover my changes.

Docs:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by alexellis** [comment]
> How Has This Been Tested?

Could you show the specific commands you ran and some logs to show the new version was running?

Please also deploy and invoke functions, then show the logs.

File: cloud-config.txt [hunk 1]
```diff
@@ -10,8 +10,8 @@ packages:
  - git
 
 runcmd:
-- curl -sLSf https://github.com/containerd/containerd/releases/download/v1.5.4/containerd-1.5.4-linux-amd64.tar.gz > /tmp/containerd.tar.gz && tar -xvf /tmp/containerd.tar.gz -C /usr/local/bin/ --strip-components=1
-- curl -SLfs https://raw.githubusercontent.com/containerd/containerd/v1.5.4/containerd.service | tee /etc/systemd/system/containerd.service
+- curl -sLSf https://github.com/containerd/containerd/releases/download/v1.5.7/containerd-1.5.-7linux-amd64.tar.gz > /tmp/containerd.tar.gz && tar -xvf /tmp/containerd.tar.gz -C /usr/local/bin/ --strip-components=1
```
Comment (alexellis) [inline cloud-config.txt:13]: This is a typo.

Did you do a global search and replace? How did you test this?

I don't think you did test it, so can you copy/paste this into a cloud control panel on DigitalOcean etc and prove that it works?

File: docs/DEV.md [hunk 2]
```diff
@@ -100,7 +100,7 @@ containerd -version
     Building `containerd` on armhf is extremely slow, so I've provided binaries for you.
 
     ```bash
-    curl -sSL https://github.com/alexellis/containerd-armhf/releases/download/v1.5.4/containerd.tgz | sudo tar -xvz --strip-components=2 -C /usr/local/bin/
+    curl -sSL https://github.com/alexellis/containerd-armhf/releases/download/v1.5.7/containerd.tgz | sudo tar -xvz --strip-components=2 -C /usr/local/bin/
```
Comment (alexellis) [inline docs/DEV.md:103]:
```
curl -sSL https://github.com/alexellis/containerd-armhf/releases/download/v1.5.7/containerd.tgz

Not Found
```

This was not tested

File: docs/bootstrap/cloud-config.tpl [hunk 3]
```diff
@@ -16,8 +16,8 @@ runcmd:
 - mkdir -p /var/lib/faasd/secrets/
 - echo ${gw_password} > /var/lib/faasd/secrets/basic-auth-password
 - echo admin > /var/lib/faasd/secrets/basic-auth-user
-- cd /go/src/github.com/openfaas/ && git clone --depth 1 --branch 0.13.0 https://github.com/openfaas/faasd
-- curl -fSLs "https://github.com/openfaas/faasd/releases/download/0.13.0/faasd" --output "/usr/local/bin/faasd" && chmod a+x "/usr/local/bin/faasd"
+- cd /go/src/github.com/openfaas/ && git clone --depth 1 --branch 0.14.2 https://github.com/openfaas/faasd
```
Comment (alexellis) [inline docs/bootstrap/cloud-config.tpl:19]: The latest is actually 0.14.3

https://github.com/openfaas/faasd/releases

File: docs/bootstrap/cloud-config.tpl [hunk 4]
```diff
@@ -16,8 +16,8 @@ runcmd:
...
+- curl -fSLs "https://github.com/openfaas/faasd/releases/download/0.14.2/faasd" --output "/usr/local/bin/faasd" && chmod a+x "/usr/local/bin/faasd"
```
Comment (alexellis) [inline docs/bootstrap/cloud-config.tpl:20]: Same as above

**Comment by ajayk** [comment]
> Thanks for volunteering to help. Unfortunately, this PR showed a lack of testing which surprised me. I've left some comments for you and hope you can turn this around.

My bad most of the changes were left out in the multipass , let me check them with the local testing results on a ubuntu box

**Comment by alexellis** [comment]
Can you use the video suggested by Derek to rebase your commits into one please?

File: hack/install.sh [hunk 5]
```diff
@@ -86,21 +86,21 @@ install_containerd() {
   arch=$(uname -m)
   case $arch in
   x86_64 | amd64)
-    curl -sLSf https://github.com/containerd/containerd/releases/download/v1.5.4/containerd-1.5.4-linux-amd64.tar.gz | $SUDO tar -xvz --strip-components=1 -C /usr/local/bin/
+    curl -sLSf https://github.com/containerd/containerd/releases/download/v1.5.7/containerd-1.5.7-linux-amd64.tar.gz | $SUDO tar -xvz --strip-components=1 -C /usr/local/bin/
```
Comment (alexellis) [inline hack/install.sh:89]: @jsiebens do you think it makes sense to extract a variable `CONTAINERD_VER` here to make maintenance easier?

**Comment by ajayk** [comment]
```
ubuntu@primary:~/Home/dev/oss/faasd$ make
CGO_ENABLED=0 GOOS=linux go test -mod=vendor -ldflags "-s -w -X main.Version=0.14.3-11-gde20d83 -X main.GitCommit=de20d83eed5d130230a4393f986c3f8dfd73d57a" ./...
?   	github.com/openfaas/faasd	[no test files]
?   	github.com/openfaas/faasd/cmd	[no test files]
ok  	github.com/openfaas/faasd/pkg	(cached)
ok  	github.com/openfaas/faasd/pkg/cninetwork	(cached)
ok  	github.com/openfaas/faasd/pkg/depgraph	(cached)
ok  	github.com/openfaas/faasd/pkg/logs	(cached)
ok  	github.com/openfaas/faasd/pkg/provider/config	(cached)
ok  	github.com/openfaas/faasd/pkg/provider/handlers	(cached)
?   	github.com/openfaas/faasd/pkg/service	[no test files]
?   	github.com/openfaas/faasd/pkg/systemd	[no test files]
CGO_ENABLED=0 GOOS=linux go build -mod=vendor -ldflags "-s -w -X main.Version=0.14.3-11-gde20d83 -X main.GitCommit=de20d83eed5d130230a4393f986c3f8dfd73d57a" -a -installsuffix cgo -o bin/faasd
CGO_ENABLED=0 GOOS=linux GOARCH=arm GOARM=7 go build -mod=vendor -ldflags "-s -w -X main.Version=0.14.3-11-gde20d83 -X main.GitCommit=de20d83eed5d130230a4393f986c3f8dfd73d57a" -a -installsuffix cgo -o bin/faasd-armhf
CGO_ENABLED=0 GOOS=linux GOARCH=arm64 go build -mod=vendor -ldflags "-s -w -X main.Version=0.14.3-11-gde20d83 -X main.GitCommit=de20d83eed5d130230a4393f986c3f8dfd73d57a" -a -installsuffix cgo -o bin/faasd-arm64
for f in bin/faasd*; do shasum -a 256 $f > $f.sha256; done


export VER=1.5.7
curl -sSL https://github.com/containerd/containerd/releases/download/v$VER/containerd-$VER-linux-amd64.tar.gz > /tmp/containerd.tar.gz \
  && sudo tar -xvf /tmp/containerd.tar.gz -C /usr/local/bin/ --strip-components=1

ubuntu@primary:~/Home/dev/oss/faasd$ containerd --version
containerd github.com/containerd/containerd v1.5.7 8686ededfc90076914c5238eb96c883ea093a8ba


ubuntu@primary:~/Home/dev/oss/faasd$ journalctl -u faasd -f
-- Logs begin at Wed 2021-11-03 18:27:36 IST. --
Nov 03 22:51:46 primary faasd[22170]: 2021/11/03 22:51:46 Resolver: "faasd-provider"="10.62.0.1"
Nov 03 22:51:46 primary faasd[22170]: 2021/11/03 22:51:46 Resolver: "basic-auth-plugin"="10.62.0.17"
Nov 03 22:51:46 primary faasd[22170]: 2021/11/03 22:51:46 Resolver: "nats"="10.62.0.18"
Nov 03 22:51:46 primary faasd[22170]: 2021/11/03 22:51:46 Resolver: "prometheus"="10.62.0.19"
Nov 03 22:51:46 primary faasd[22170]: 2021/11/03 22:51:46 Resolver: "gateway"="10.62.0.20"
Nov 03 22:51:46 primary faasd[22170]: 2021/11/03 22:51:46 Resolver: "queue-worker"="10.62.0.21"
Nov 03 22:51:46 primary faasd[22170]: 2021/11/03 22:51:46 Looking up IP for: "prometheus"
Nov 03 22:51:46 primary faasd[22170]: 2021/11/03 22:51:46 Proxy from: 127.0.0.1:9090, to: prometheus:9090 (10.62.0.19)
Nov 03 22:51:46 primary faasd[22170]: 2021/11/03 22:51:46 faasd: waiting for SIGTERM or SIGINT
Nov 03 22:51:46 primary faasd[22170]: 2021/11/03 22:51:46 Proxy from: 0.0.0.0:8080, to: gateway:8080 (10.62.0.20)
^C
ubuntu@primary:~/Home/dev/oss/faasd$ journalctl -u faasd-provider -f
-- Logs begin at Wed 2021-11-03 18:27:36 IST. --
Nov 03 23:18:08 primary faasd[22166]: 2021/11/03 23:18:08 curl has IP: 10.62.0.22.
Nov 03 23:18:18 primary faasd[22166]: 2021/11/03 23:18:18 Resolve: "curl.openfaas-fn"
Nov 03 23:18:18 primary faasd[22166]: 2021/11/03 23:18:18 curl.openfaas-fn took 0.023719 seconds
Nov 03 23:19:19 primary faasd[22166]: 2021/11/03 23:19:19 [Deploy] request: {"image":"ghcr.io/openfaas/alpine:latest","service":"env","envProcess":"env","namespace":"openfaas-fn","envVars":{},"secrets":[],"labels":{},"annotations":{}}
Nov 03 23:19:23 primary faasd[22166]: 2021/11/03 23:19:23 [Deploy] request: {"image":"ghcr.io/openfaas/alpine:latest","service":"env","envProcess":"env","namespace":"openfaas-fn","envVars":{},"secrets":[],"labels":{},"annotations":{}}
Nov 03 23:19:24 primary faasd[22166]: 2021/11/03 23:19:24 Image for: ghcr.io/openfaas/alpine:latest size: 6178097, took: 5.008815s
Nov 03 23:19:24 primary faasd[22166]: 2021/11/03 23:19:24 Image for: ghcr.io/openfaas/alpine:latest size: 6178097, took: 1.100048s
Nov 03 23:19:24 primary faasd[22166]: 2021/11/03 23:19:24 [Deploy] error deploying env, error: unable to create container: env, error: snapshot "env-snapshot": already exists
Nov 03 23:19:24 primary faasd[22166]: 2021/11/03 23:19:24 Container ID: env        Task ID env:        Task PID: 23024
Nov 03 23:19:25 primary faasd[22166]: 2021/11/03 23:19:25 env has IP: 10.62.0.23.
^C
ubuntu@primary:~/Home/dev/oss/faasd$ faasd version
  __                     _ 
 / _| __ _  __ _ ___  __| |
| |_ / _` |/ _` / __|/ _` |
|  _| (_| | (_| \__ \ (_| |
|_|  \__,_|\__,_|___/\__,_|

faasd version: 0.14.3-11-gde20d83	commit: de20d83eed5d130230a4393f986c3f8dfd73d57a


ubuntu@primary:~/Home/dev/oss/faasd$ journalctl -u containerd -f
-- Logs begin at Wed 2021-11-03 18:27:36 IST. --
Nov 03 23:20:59 primary openfaas:basic-auth-plugin[22237]: 2021/11/03 17:50:59 Validated request 200.
Nov 03 23:20:59 primary openfaas:gateway[22520]: 2021/11/03 17:50:59 Forwarded [GET] to /system/function/env?namespace=openfaas-fn - [200] - 0.027264s seconds
Nov 03 23:20:59 primary openfaas:gateway[22520]: 2021/11/03 17:50:59 Forwarded [GET] to /system/functions?namespace=openfaas-fn - [200] - 0.046338s seconds
Nov 03 23:21:01 primary openfaas:basic-auth-plugin[22237]: 2021/11/03 17:51:01 Validated request 200.
Nov 03 23:21:01 primary openfaas:gateway[22520]: 2021/11/03 17:51:01 Forwarded [GET] to /system/function/env?namespace=openfaas-fn - [200] - 0.016864s seconds
Nov 03 23:21:03 primary openfaas:basic-auth-plugin[22237]: 2021/11/03 17:51:03 Validated request 200.
Nov 03 23:21:03 primary openfaas:gateway[22520]: 2021/11/03 17:51:03 Forwarded [GET] to /system/functions?namespace=openfaas-fn - [200] - 0.019800s seconds
Nov 03 23:21:04 primary openfaas:basic-auth-plugin[22237]: 2021/11/03 17:51:04 Validated request 200.
Nov 03 23:21:04 primary openfaas:gateway[22520]: 2021/11/03 17:51:04 Forwarded [GET] to /system/function/env?namespace=openfaas-fn - [200] - 0.028419s seconds
Nov 03 23:21:06 primary openfaas:basic-auth-plugin[22237]: 2021/11/03 17:51:06 Validated request 200.
Nov 03 23:21:06 primary openfaas:basic-auth-plugin[22237]: 2021/11/03 17:51:06 Validated request 200.
Nov 03 23:21:06 primary openfaas:gateway[22520]: 2021/11/03 17:51:06 Forwarded [GET] to /system/function/env?namespace=openfaas-fn - [200] - 0.040742s seconds
Nov 03 23:21:06 primary openfaas:gateway[22520]: 2021/11/03 17:51:06 Forwarded [GET] to /system/functions?namespace=openfaas-fn - [200] - 0.070394s seconds
Nov 03 23:21:08 primary openfaas:basic-auth-plugin[22237]: 2021/11/03 17:51:08 Validated request 200.
Nov 03 23:21:08 primary openfaas:gateway[22520]: 2021/11/03 17:51:08 Forwarded [GET] to /system/function/env?namespace=openfaas-fn - [200] - 0.015555s seconds
Nov 03 23:21:10 primary openfaas:basic-auth-plugin[22237]: 2021/11/03 17:51:10 Validated request 200.
Nov 03 23:21:10 primary openfaas:gateway[22520]: 2021/11/03 17:51:10 Forwarded [GET] to /system/functions?namespace=openfaas-fn - [200] - 0.039748s seconds
Nov 03 23:21:11 primary openfaas:basic-auth-plugin[22237]: 2021/11/03 17:51:11 Validated request 200.
Nov 03 23:21:11 primary openfaas:gateway[22520]: 2021/11/03 17:51:11 Forwarded [GET] to /system/function/env?namespace=openfaas-fn - [200] - 0.020270s seconds
Nov 03 23:21:13 primary openfaas:basic-auth-plugin[22237]: 2021/11/03 17:51:13 Validated request 200.
Nov 03 23:21:13 primary openfaas:gateway[22520]: 2021/11/03 17:51:13 Forwarded [GET] to /system/functions?namespace=openfaas-fn - [200] - 0.032034s seconds
Nov 03 23:21:14 primary openfaas:basic-auth-plugin[22237]: 2021/11/03 17:51:14 Validated request 200.
Nov 03 23:21:14 primary openfaas:gateway[22520]: 2021/11/03 17:51:14 Forwarded [GET] to /system/function/env?namespace=openfaas-fn - [200] - 0.017866s seconds
Nov 03 23:21:16 primary openfaas:basic-auth-plugin[22237]: 2021/11/03 17:51:16 Validated request 200.
Nov 03 23:21:16 primary openfaas:basic-auth-plugin[22237]: 2021/11/03 17:51:16 Validated request 200.
Nov 03 23:21:16 primary openfaas:gateway[22520]: 2021/11/03 17:51:16 Forwarded [GET] to /system/function/env?namespace=openfaas-fn - [200] - 0.029181s seconds
Nov 03 23:21:16 primary openfaas:gateway[22520]: 2021/11/03 17:51:16 Forwarded [GET] to /system/functions?namespace=openfaas-fn - [200] - 0.038267s seconds
Nov 03 23:21:18 primary openfaas:basic-auth-plugin[22237]: 2021/11/03 17:51:18 Validated request 200.
Nov 03 23:21:18 primary openfaas:gateway[22520]: 2021/11/03 17:51:18 Forwarded [GET] to /system/function/env?namespace=openfaas-fn - [200] - 0.016346s seconds
Nov 03 23:21:18 primary openfaas:gateway[22520]: 2021/11/03 17:51:18 Get "http://faasd-provider:8081/system/namespaces": dial tcp: i/o timeout
^C



ubuntu@primary:~/Home/dev/oss/faasd$ journalctl -u faasd-provider -f
-- Logs begin at Wed 2021-11-03 18:27:36 IST. --
Nov 03 23:24:45 primary faasd[22166]: 2021/11/03 23:24:45 Image for: ghcr.io/openfaas/nodeinfo:latest size: 35215692, took: 21.657420s
Nov 03 23:24:45 primary faasd[22166]: 2021/11/03 23:24:45 [Deploy] error deploying nodeinfo, error: unable to create container: nodeinfo, error: snapshot "nodeinfo-snapshot": already exists
Nov 03 23:24:49 primary faasd[22166]: 2021/11/03 23:24:49 Image for: docker.io/functions/hubstats:latest size: 12948112, took: 14.957942s
Nov 03 23:24:49 primary faasd[22166]: 2021/11/03 23:24:49 Resolve: "nodeinfo.openfaas-fn"
Nov 03 23:24:50 primary faasd[22166]: 2021/11/03 23:24:50 nodeinfo.openfaas-fn took 0.533208 seconds
Nov 03 23:24:51 primary faasd[22166]: 2021/11/03 23:24:51 Container ID: hubstats        Task ID hubstats:        Task PID: 23391
Nov 03 23:24:52 primary faasd[22166]: 2021/11/03 23:24:52 hubstats has IP: 10.62.0.26.
Nov 03 23:25:11 primary faasd[22166]: 2021/11/03 23:25:11 Image for: ghcr.io/openfaas/cows:latest size: 47023566, took: 32.492600s
Nov 03 23:25:12 primary faasd[22166]: 2021/11/03 23:25:12 Container ID: cows        Task ID cows:        Task PID: 23499
Nov 03 23:25:12 primary faasd[22166]: 2021/11/03 23:25:12 cows has IP: 10.62.0.27.
```


```
➜  faasd git:(containerd-1.5.7) multipass list 
Name                    State             IPv4             Image
primary                 Running           192.168.64.4     Ubuntu 20.04 LTS
                                          10.62.0.1
foo                     Running           192.168.64.3     Ubuntu 20.04 LTS
```
Screen Shot 2021-11-03 at 11 52 47 PM

**Comment by alexellis** [comment]
I'm going to move this over to 1.6.x thanks for the PR.

**Pull request closed without merging**